### ISSUE TITLE
Centralize Access to `_googlesitekitModulesData` Global

### ISSUE THREAD

**Issue by zutigrm** [google/site-kit-wp#9573]
## Feature Description

`_googlesitekitModulesData` is accessed from different selector/resolvers across the codebase, we should centralise this check potentially with new selector/resolver, and re-use it instead of checking it directly against global in several places.

Check this [PR comment](https://github.com/google/site-kit-wp/pull/9487#discussion_r1819534731) for more details

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* `assets/js/googlesitekit/modules/datastore/modules.js` is updated to include action|resolver|selector to retrieve data from `_googlesitekitModulesData` and store it to the state
* Existing resolvers, selectors and actions added individually in current modules are refactored, so only the new global resolver/selector is called

## Implementation Brief
* [ ] Update `assets/js/googlesitekit/modules/datastore/modules.js`
  * Add `inlineModulesData` to the `baseInitialState` (initially `undefined`)
  * Create `receiveInlineModulesData` action with proper validation which yields data to the reducer
  * Add reducer case to store the data received from previous action in the state and update `inlineModulesData` value
  * Create `getInlineModulesData` resolver that reads from `global._googlesitekitModulesData` and forwards the data to the `receiveInlineModulesData` action
  * Create `getInlineModulesData` selector to access the stored data
  * Create `getModuleInlineData(module_slug)` selector to extract data for specific modules
* [ ] Refactor existing individual resolvers that directly access `global._googlesitekitModulesData`, to use the new global `getModuleInlineData(module_slug)` selector in:
  * `assets/js/modules/analytics-4/datastore/module-data.js`
  * `assets/js/modules/ads/datastore/module-data.js`
  * `assets/js/modules/sign-in-with-google/datastore/module-data.js` 
  * `assets/js/modules/analytics-4/datastore/custom-dimensions-gathering-data.js`
  * `assets/js/modules/analytics-4/datastore/partial-data.js`
  * `assets/js/modules/sign-in-with-google/components/setup/SetupForm.js`

### Test Coverage

* Update `assets/js/googlesitekit/modules/datastore/modules.test.js` to include basic coverage for new resolver/selector and action

## QA Brief (QA:ENG)

* Set up Site Kit with Analytics, Ads, and Sign In With Google.  
* Make sure each module’s selector data matches the corresponding global object’s data: 
	* The result of `googlesitekit.data.select("modules/analytics-4").getModuleData()` should match the values in ` _googlesitekitModulesData['analytics-4']`
	* The result of `googlesitekit.data.select("modules/ads").getModuleData()` should match the values in ` _googlesitekitModulesData['ads']`
	* The result of `googlesitekit.data.select("modules/sign-in-with-google").getModuleData()` should match the values in `_googlesitekitModulesData['sign-in-with-google']`

## Changelog entry

* N/A.

**Comment by techanvil** [comment]
AC ✅

**Comment by tofumatt** [comment]
IB ✅

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

- Tested on dev environment.
- Verified console code results for all 3 modules: Analytics, Ads, and SIWG.
- Made changes in the settings and confirmed that results update correctly in both calls.

@abdelmalekkkkk I have a question regarding the Analytics module values:
When running `googlesitekit.data.select("modules/analytics-4").getModuleData`(), I see an additional value `hasMismatchTag`, which does not exist in the response of `_googlesitekitModulesData['analytics-4'`].
Is this expected?

<img width="1645" height="632" alt="Image" src="https://github.com/user-attachments/assets/96d73fd5-bb01-4ecb-9303-fea0227c1105" />

**PASS CASES**



**Analytics module**

`googlesitekit.data.select("modules/analytics-4").getModuleData() `and `_googlesitekitModulesData['analytics-4']`

<details>

<img width="1621" height="618" alt="Image" src="https://github.com/user-attachments/assets/911d49fb-23f8-43ba-bd99-46c4ae1bdaec" />

</details>

**Ads module**

`googlesitekit.data.select("modules/ads").getModuleData()` and `_googlesitekitModulesData['ads']`

<details>

<img width="1616" height="498" alt="Image" src="https://github.com/user-attachments/assets/496a6e38-b817-4840-8934-04e36cdd84d2" />

After enabling WooCommerce plugin ( values matching)

<img width="1634" height="557" alt="Image" src="https://github.com/user-attachments/assets/8bae1f40-104a-42b2-9551-a8f107bc5767" />

</details>

**SIWG Module**

`googlesitekit.data.select("modules/sign-in-with-google").getModuleData()` and `_googlesitekitModulesData['sign-in-with-google']`
<details>

<img width="1654" height="364" alt="Image" src="https://github.com/user-attachments/assets/b3e733ab-a86a-4d8c-ab27-73dec9bf3171" />


</details>

**Comment by abdelmalekkkkk** [comment]
@mohitwp Yes, that's expected. It's just the `tagIDMismatch` field being renamed to `hasMismatchTag`.

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on dev environment.
- Verified console code results for all 3 modules: Analytics, Ads, and SIWG.
- Made changes in the settings and confirmed that results update correctly in both calls.
- Verified that values are matching for all modules.



**Analytics module**

`googlesitekit.data.select("modules/analytics-4").getModuleData() `and `_googlesitekitModulesData['analytics-4']`

<details>

<img width="1621" height="618" alt="Image" src="https://github.com/user-attachments/assets/911d49fb-23f8-43ba-bd99-46c4ae1bdaec" />

</details>

**Ads module**

`googlesitekit.data.select("modules/ads").getModuleData()` and `_googlesitekitModulesData['ads']`

<details>

<img width="1616" height="498" alt="Image" src="https://github.com/user-attachments/assets/496a6e38-b817-4840-8934-04e36cdd84d2" />

After enabling WooCommerce plugin ( values matching)

<img width="1634" height="557" alt="Image" src="https://github.com/user-attachments/assets/8bae1f40-104a-42b2-9551-a8f107bc5767" />

</details>

**SIWG Module**

`googlesitekit.data.select("modules/sign-in-with-google").getModuleData()` and `_googlesitekitModulesData['sign-in-with-google']`
<details>

<img width="1654" height="364" alt="Image" src="https://github.com/user-attachments/assets/b3e733ab-a86a-4d8c-ab27-73dec9bf3171" />


</details>
